fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/mappers/DateTimeSortMapper.java

```diff
@@ -0,0 +1,78 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.metric.mappers;
+
+import com.yahoo.bard.webservice.data.Result;
+import com.yahoo.bard.webservice.data.ResultSet;
+import com.yahoo.bard.webservice.druid.model.orderby.SortDirection;
+import com.yahoo.bard.webservice.table.Schema;
+
+import org.joda.time.DateTime;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ *  Mapper to sort the result set based on dateTime column sort direction.
+ */
+public class DateTimeSortMapper extends ResultSetMapper {
+
+    private SortDirection direction;
+
+    /**
+     * Constructor.
+     *
+     * @param direction  Sort direction
+     */
+    public DateTimeSortMapper(SortDirection direction) {
+        this.direction = direction;
+    }
+
+    /**
+     *  Sorting the resultSet based on dateTime column sort direction.
+     *
+     * @param resultSet  The result set need to be sorted in ascending or descending order
+     *
+     * @return sorted ResultSet
+     */
+    @Override
+    public ResultSet map(ResultSet resultSet) {
```

Comment:
Can we wrap a RequestLog timer around this, so we can get a sense of how long sorting is taking?